### PR TITLE
feat(engine): Turn workflow into API

### PR DESCRIPTION
**Pull request by mattdurant**
Webhook endpoint for synchronous execution and return final workflow output.

## Description

This is so that you can call a webhook in Tracecat to perform some actions and receive as a response the final_context object from the workflow.

My current use case is an outgoing webhook in Teams instead of a full bot implementation, so I can send a message in a channel to @ChatOps with a command, parse the command in the workflow and then do some work and the final_context is returned.  In this case, a chat message indicating success or failure.

## Related Tickets & Documents
N/A

## Screenshots/Recordings

N/A

## Steps to QA

Test by calling a webhook, but add /wait to the URL, the response will return once the workflow is finished and contain the final_context of the workflow.

### CODE REVIEW

File: tracecat/api/app.py [hunk 1]
```diff
@@ -373,6 +373,43 @@ async def incoming_webhook(
     return response
 
 
+@app.post("/webhooks/{path}/{secret}/wait", tags=["public"])
+async def incoming_webhook_Wait(
```
Comment (daryllimyt) [inline tracecat/api/app.py]: style: snake case

File: tracecat/api/app.py [hunk 2]
```diff
@@ -373,6 +373,43 @@ async def incoming_webhook(
...
+    path: str,
+    payload: dict[str, Any] | None = None,
+    x_tracecat_enable_runtime_tests: Annotated[str | None, Header()] = None,
+) -> dict[str, Any]:
+    """
+    Webhook endpoint to trigger a workflow.
+
+    This is an external facing endpoint is used to trigger a workflow by sending a webhook request.
+    The workflow is identified by the `path` parameter, which is equivalent to the workflow id.
+    """
```
Comment (daryllimyt) [inline tracecat/api/app.py]: could we update the docstring here to reflect that this endpoint behaves differently from the regular one?

File: tracecat/api/app.py (hunk 1 above)
Comment (topher-lo) [inline tracecat/api/app.py]:
```suggestion
async def incoming_webhook_wait(
```

**Review by daryllimyt — CHANGES_REQUESTED** [review]

File: tracecat/api/app.py (hunk 1 above)
Comment (mattdurant) [reply tracecat/api/app.py]: Out of curiosity, why would wait be capitalized here? I've not seen snake case have any words capitalized like this before.

Comment (mattdurant) [reply tracecat/api/app.py]: Oh man, ignore me, I completely misread Chris' review and thought he suggested capitalizing Wait.  I didn't realize I committed it that way...

Comment (daryllimyt) [reply tracecat/api/app.py]: no worries :)

**Review by daryllimyt — APPROVED** [review]
LGTM

**Comment by topher-lo** [comment]
Will delay this PR for next week's release while we rework the API routers and add tests this week.

**Review by topher-lo — COMMENTED** [review]
Wait for API work to complete this week

**Comment by mattdurant** [comment]
@daryllimyt @topher-lo based on the conflict here, I think you guys did a big refactoring during the same time, I think I should close this PR and start clean make the 2-3 changes to enable this again and re-submit, agreed?

**Comment by daryllimyt** [comment]
> @daryllimyt @topher-lo based on the conflict here, I think you guys did a big refactoring during the same time, I think I should close this PR and start clean make the 2-3 changes to enable this again and re-submit, agreed?

Yes - I'm also aiming to close #285  so I think it makes sense to adjust it accordingly

**Pull request closed without merging**